src/py/flwr/client/process/process.py

```diff
@@ -0,0 +1,122 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Flower background ClientApp."""
+
+from logging import DEBUG, ERROR, INFO
+from typing import Tuple
+
+import grpc
+
+from flwr.client.client_app import ClientApp
+from flwr.common import Context, Message
+from flwr.common.grpc import create_channel
+from flwr.common.logger import log
+from flwr.common.serde import (
+    context_from_proto,
+    context_to_proto,
+    message_from_proto,
+    message_to_proto,
+    run_from_proto,
+)
+from flwr.common.typing import Run
+
+# pylint: disable=E0611
+from flwr.proto.clientappio_pb2 import (
+    PullClientAppInputsRequest,
+    PullClientAppInputsResponse,
+    PushClientAppOutputsRequest,
+    PushClientAppOutputsResponse,
+)
+from flwr.proto.clientappio_pb2_grpc import ClientAppIoStub
+
+from .utils import _get_load_client_app_fn
+
+
+def _run_background_client(  # pylint: disable=R0914
```

Comment:
Why does this start with an `_`? We use it in a different module

Reply:
Good question. We also use it elsewhere, e.g. `_start_client_internal` (called in `supernode/app.py`, defined in `client/app.py`). I assumed this is to discourage users from using this 'private' API.

Re: `background`. I agree this can be dropped since this function is not just for background process.

Reply:
Changed it to `_run_clientapp`.

Reply:
Prefix will be removed in #4026